toxcore/query.h

```diff
@@ -0,0 +1,57 @@
+/** query.h
+ *
+ * Makes requests for information using the DHT or Onion as appropriate.
+ */
+
+#ifndef TOX_QUERY_H
+#define TOX_QUERY_H
+
+#include "network.h"
+#include "tox.h"
```

Comment:
We need to talk about this. Traditionally, we don't include headers from layers above us. Perhaps you can suggest a way to achieve your goal without introducing (additional) cyclic dependencies?